virttest/test_setup.py

```diff
@@ -2423,38 +2424,71 @@ def __init__(self, test, log_level="1", log_file=""):
+                "backupfile": "%s.backup" % utils_config.VirtProxydConfig().conf_path}
+            self.daemons_dict["virtstoraged"] = {
+                "daemon": utils_libvirtd.Libvirtd("virtstoraged"),
+                "conf": utils_config.VirtStoragedConfig(),
+                "backupfile": "%s.backup" % utils_config.VirtStoragedConfig().conf_path}
+            self.daemons_dict["virtinterfaced"] = {
+                "daemon": utils_libvirtd.Libvirtd("virtinterfaced"),
+                "conf": utils_config.VirtInterfacedConfig(),
+                "backupfile": "%s.backup" % utils_config.VirtInterfacedConfig().conf_path}
+            self.daemons_dict["virtnodedevd"] = {
+                "daemon": utils_libvirtd.Libvirtd("virtnodedevd"),
+                "conf": utils_config.VirtNodedevdConfig(),
+                "backupfile": "%s.backup" % utils_config.VirtNodedevdConfig().conf_path}
+            self.daemons_dict["virtnwfilterd"] = {
+                "daemon": utils_libvirtd.Libvirtd("virtnwfilterd"),
+                "conf": utils_config.VirtNwfilterdConfig(),
+                "backupfile": "%s.backup" % utils_config.VirtNwfilterdConfig().conf_path}
 
     def enable(self):
         """ Enable libvirtd debug log """
         if not self.log_file or not os.path.isdir(os.path.dirname(self.log_file)):
             self.log_file = utils_misc.get_path(self.test.debugdir,
                                                 "libvirtd.log")
-        if os.path.isfile(self.backupfile):
-            os.remove(self.backupfile)
-
-        # backup libvirtd conf before restarting libvirtd
-        try:
-            with open(self.backupfile, "w") as fd:
-                fd.write(self.libvirtd_conf.backup_content)
-                fd.close()
-        except IOError as info:
-            self.test.error(info)
-
         # param used during libvirtd cleanup
         self.test.params["libvirtd_debug_file"] = self.log_file
         logging.debug("libvirtd debug log stored in: %s", self.log_file)
-        self.libvirtd_conf["log_level"] = self.log_level
-        self.libvirtd_conf["log_outputs"] = '"%s:file:%s"' % (self.log_level,
-                                                              self.log_file)
-        self.libvirtd.restart()
+
+        for value in self.daemons_dict.values():
+            if os.path.isfile(value.get("backupfile")):
+                os.remove(value.get("backupfile"))
+
+            # backup config files before restarting services
+            try:
+                with open(value.get("backupfile"), "w") as fd:
+                    fd.write(value.get("conf").backup_content)
+                    fd.close()
+            except IOError as info:
+                self.test.error(info)
+
+            value.get("conf")["log_level"] = self.log_level
+            value.get("conf")["log_outputs"] = '"%s:file:%s"' % (self.log_level,
+                                                                 self.log_file)
```

Comment:
It is ok to just share same 'self.log_level' among all sub daemons, but for self.log_file, it seems not very reasonable.

Reply:
Actually, this is to continue the previous design which keeps libvirtd.log if test fails or errors. I've talked about feature owner, at this stage, they prefer it.

Reply:
Ok. Let's see if there is an issue after some tests.